internal/infrastructure/kubernetes/infra.go

```diff
@@ -66,6 +67,10 @@ func (i *Infra) createOrUpdate(ctx context.Context, r ResourceRender) error {
 		return fmt.Errorf("failed to create or update deployment %s/%s: %w", i.Namespace, r.Name(), err)
 	}
 
+	if err := i.createOrUpdateDaemonSet(ctx, r); err != nil {
```

Comment:
Maybe we should be more cautious here and make DS/DP creation mutually exclusive, even if we have CEL validations that protect from both DS and DP being configured? @arkodg  - WDYT? I know that we sometimes prefer to not rely only on CEL.

Reply:
Besides CEL validation, there is also a check for which type of deployment (Deployment or DaemonSet) is used [HERE](https://github.com/zvlb/gateway/blob/ds/api/v1alpha1/envoyproxy_helpers.go#L63). So, if Deployment is not defined (DaemonSet is used), nil is returned, and if Deployment was previously used, it will be deleted. And vice versa